### DISCUSSION TITLE
useSnapshot render tow times per state change

### DISCUSSION THREAD

**Discussion by wenerme** [pmndrs/valtio#513] [Q&A] [answered]
Do any state change action will cause the render count +2
https://codesandbox.io/s/valtio-render-test-lzqhf7

I expect only one render per change.

**Comment by dai-shi** [comment, marked as answer]
This is something `<StrictMode>` does. Unless it's causing real problem for you, it's fine. If there's something wrong, it means `StrictMode` reveals some potential bugs/misusages.

Does this work as expected? If it's not what you are describing, please let us know.
